docs/side_quests/nf-test.md

```diff
@@ -0,0 +1,800 @@
+
+[**nf-test**](https://www.nf-test.com/) is a tool that allows you to write module, workflow and pipeline level test. In short, it allows you to systematically check every individual part of the pipeline is working as expected, _in isolation_.
+
+In this part of the training, we're going to show you how to use nf-test to write module-level tests for the three processes in our pipeline.
+
+---
+
+## 0. Warmup
+
+Let's move into the project directory.
+
+```bash
+cd side-quests/nf-test
+```
+
+The `nf-test` directory has the file content like:
+
+```console title="Directory contents"
+nf-test
+├── greetings.csv
+├── main.nf
+└── nextflow.config
+```
+
+For a detailed description of the files, see the [warmup from Hello Nextflow](../hello_nextflow/00_orientation.md).
+
+### 0.1. Run the workflow
+
+Let's run the workflow to make sure it's working as expected.
+
+```bash
+nextflow run main.nf
+```
+
+CONGRATULATIONS! You just ran a unit test!
+
+"Wait, what? I just ran the workflow and it worked! How is that a unit test?"
+
+Good question!
+
+Let's break down what just happened.
+
+You ran the workflow with the default parameters, you confirmed it worked and you're happy with the results. This is the essence of unit testing. As you go through this course you will notice we always run the workflow at the start to confirm everything is set up correctly. This gives us the confidence to continue with the training, knowing the code started in a working state.
+
+Unit testing software essentially does this process for us. Let's replace our simple `nextflow run main.nf` with a standardised test provided by nf-test.
+
+### Takeaway
+
+You should be able to 'test' a pipeline by manually running it.
+
+### What's next?
+
+Initialize `nf-test`.
+
+---
+
+## 1.0. Initialize `nf-test`
+
+The `nf-test` package provides an initialization command that sets up a few things in order for us to start developing tests for our project.
+
+```bash
+nf-test init
+```
+
+This should produce the following output:
+
+```bash
+🚀 nf-test 0.9.1
+https://www.nf-test.com
+(c) 2021 - 2024 Lukas Forer and Sebastian Schoenherr
+
+Project configured. Configuration is stored in nf-test.config
+```
+
+It also creates a `tests` directory containing a configuration file stub.
+
+## 1.1. Generate an nf-test
+
+`nf-test` comes with a set of tools for building nf-test files, saving us the majority of the work. These come under the subcommand `generate`. Let's generate a test for the pipeline:
+
+```bash
+nf-test generate pipeline main.nf
+```
+
+```console title="Output"
+> nf-test generate pipeline main.nf
+
+Load source file '/workspaces/training/side-quests/nf-test/main.nf'
+Wrote pipeline test file '/workspaces/training/side-quests/nf-test/tests/main.nf.test
+
+SUCCESS: Generated 1 test files.
+```
+
+This will create a `tests` directory with a `main.nf.test` and a `nextflow.config` file. This is our pipeline level test file. If you run `tree tests/` you should see something like this:
```

Comment:
does nf-test create the tests/ folder?
I think I noticed a different behavior: https://github.com/askimed/nf-test/issues/286

Reply:
It does